include/api/CInferenceModelDefinition.h

```diff
@@ -47,6 +47,17 @@ class API_EXPORT CSerializableToJsonStream {
     virtual void addToJsonStream(TGenericLineWriter& /*writer*/) const = 0;
 };
 
+class API_EXPORT CSerializableToJsonDocumentCompressed : public CSerializableToJsonStream{
```

Comment:
How about moving `CSerializableToJsonDocument`,  `CSerializableToJsonStream` and `CSerializableToJsonDocumentCompressed` to their own header / cc file? It feels kind of odd to have them declared here now that they usage is shared. Maybe `CSerializableToJson.h/cc`? Also I think some documentation for this class is warranted.

Reply:
I created `CSerializableToJson.h/cc` with the common code.